src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -7196,5 +7202,834 @@ static void M(ReadOnlySpan<int> s)
+        {
+            var source = """
+                class C
+                {
+                    S M1(S s1, S s2)
+                    {
+                        S s = s1 && s2;
+                        return s; // 1
+                    }
+
+                    S M2(S s1, S s2)
+                    {
+                        return s1 && s2; // 2
+                    }
+
+                    S M3(in S s1, in S s2)
+                    {
+                        S s = s1 && s2;
+                        return s;
+                    }
+
+                    S M4(scoped in S s1, in S s2)
+                    {
+                        S s = s1 && s2;
+                        return s; // 3
+                    }
+
+                    S M5(in S s1, scoped in S s2)
+                    {
+                        S s = s1 && s2;
+                        return s; // 4
+                    }
+                }
+
+                ref struct S
+                {
+                    public static bool operator true(in S s) => throw null;
+                    public static bool operator false(in S s) => throw null;
+                    public static S operator &(in S x, in S y) => throw null;
+                    public static S operator |(in S x, in S y) => throw null;
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (6,16): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
+                //         return s; // 1
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(6, 16),
+                // (11,16): error CS8166: Cannot return a parameter by reference 's1' because it is not a ref parameter
+                //         return s1 && s2; // 2
+                Diagnostic(ErrorCode.ERR_RefReturnParameter, "s1").WithArguments("s1").WithLocation(11, 16),
+                // (11,16): error CS8347: Cannot use a result of 'S.operator &(in S, in S)' in this context because it may expose variables referenced by parameter 'x' outside of their declaration scope
+                //         return s1 && s2; // 2
+                Diagnostic(ErrorCode.ERR_EscapeCall, "s1 && s2").WithArguments("S.operator &(in S, in S)", "x").WithLocation(11, 16),
+                // (23,16): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
+                //         return s; // 3
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(23, 16),
+                // (29,16): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
+                //         return s; // 4
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(29, 16));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider adding a test for an operator with either or both parameters as `scoped`. The following results in errors before this change:
```csharp
ref struct R
{
    private ref readonly int _i;
    public R(in int i) { _i = ref i; }
    public static bool operator true(R r) => true;
    public static bool operator false(R r) => false;
    public static R operator |(scoped R x, scoped R y) => default;
}
class Program
{
    static R F()
    {
        return new R(1) || new R(2);
    }
}
```